include/aspect/particle/integrator/rk_2.h

```diff
@@ -116,12 +96,9 @@ namespace aspect
           unsigned int integrator_substep;
 
           /**
-           * The particle location before the first integration step. This is
-           * used in the second step and transferred to another process if
-           * the particle leaves the domain during the first step.
+           * The location of the RK2 data that is stored in the particle properties.
            */
-          std::map<types::particle_index, Point<dim> >   loc0;
-
+          unsigned int property_location;
```

Comment:
I'm having second thoughts about the names of these variables. They represent indices, so maybe `stage_data_index_within_properties`? That's a longer name, but I think it better represents what this variable is about.

Reply:
I feel that is longer than is useful, but what about `property_index`? It reads quite well in the new code I think.